crates/mitex-parser/src/parser.rs

```diff
@@ -37,25 +35,19 @@ mod list_state {
     pub struct ListState {
         /// The checkpoint of the first item in the list
         /// Note: if an infix command is parsed, this will become None
-        start: MaybeUninit<Checkpoint>,
```

Comment:
It is not good to simply replace an Option here, since it is the mitex-parser's bottleneck of performance and stack size usage. And, our current parser should just work, because if we do wrong things on `MaybeUninit`s, mitex doesn't pass any current test at all.

But I want to remove unsafe here as well, so I'm fine with changes if we find a way to keep size of `ListState` at the same time.

Reply:
I see, because the Option discriminants are subject to alignment individually?

Reply:
One good way might be to somehow introduce a niche in Checkpoint. Not at computer right now so can't check internals but something like NonZero* for one of the fields would work.

Reply:
Unfortunately, `Checkpoint` is a type from `rowan` crate. Its definition is:
```rs
pub struct Checkpoint(usize);
```

Reply:
OK, I will open a PR on rowan to add niching to Checkpoint. That will transparently fix the sizing issue (actually, the size will decrease).

Reply:
Ok, rust-analyzer/rowan#161

Reply:
> Ok, [rust-analyzer/rowan#161](https://github.com/rust-analyzer/rowan/pull/161)

Great! I'm not sure whether your PR could be accepted, but that wouldn't block our PR. I've remove unitialized fields by some hacky way, please check https://github.com/mattfbacon/mitex/pull/2.

Reply:
The PR was merged. Can we leave it as Option in expectation of the future fix?